### DISCUSSION TITLE
Auto restart and shedulle server commands feature?

### DISCUSSION THREAD

**Discussion by KryanThePacifist** [pufferpanel/pufferpanel#1096] [Q&A]
Hello.
Have been using Puffer for my minecraft server, as opposed to cloud hosting, and this is by far the easiest solution to install and run on linux, and i love how everything just works as it should.
I do however, have a little question.
Is there a way to schedule auto restarts on servers and  issue commands on a schedule as well?
If not, could this be a feature that can be added in the future?
I messed with a couple of plugins that auto restart, and they stop the server and announce everything nice, but they won't auto start again. I assume it is either something on my config missing, or the way puffer deals with server instances makes it act like so.
Thanks for your time, and hope you have a nice weekend.

**Comment by Rajdave69** [comment]
I have a working solution. I guess you're using spigot/paper. GO in the spigot.yml and **remove** the startup script thing first. Then, click on the "Admin" option from the bottom right in your server's controls. Enable "Restart the server when it stops normally".

With that, Use UltimateAutoRestart. It works perfectly (the spigot config may start a diff instance of the server which may cause issues, that's why I mentioned that.)

**Reply by KryanThePacifist** [reply]
I appreciate the answer. I have this function turned on in my server for when the plug-in stops the server it prompts back up again no problem. however, it also means that if i stop my server manually, it will auto restart it, unless i turn the function off again to fully stop the server.
I do use UltimateAutoRestart right now as a workaround, but in the future, having this feature natively on Puffer Panel would be a plus :)

**Comment by burneykb** [comment]
I think this could be a nice feature to have natively as well.

I've been thinking about how I can schedule backups of my server that I'm self hosting for my friends. something like every day at 3am from a cron job would be more than enough. So i wrote my backup script, but I don't want to have it do a backup while the server is still running. (as I understand, this could lead to troubles). I can use RCON to stop the server inside my backup script, but then I can no longer use RCON to start the server up again from the script.

If there was a scheduler or at least some commands for starting stopping servers outside of the web interface that would be super useful! I guess I'll need to start looking at the source code. I have a feeling this will all be a bit over my head though :p

**Reply by nepcore** [reply]
The API is entirely capable of starting/stopping servers (or anything the panel can do really)

Also, a scheduling system has been in the works for a while now, so it'll happen eventually

**Reply by burneykb** [reply]
interesting! I did not see that in the documentation anywhere. Do you have any links I could check out?

**Comment by Pantheris2019** [comment]
So could I just install cron and call the pufferpanel to restart a specific server name or ID?
